comp/core/settings/component.go

```diff
@@ -32,13 +32,18 @@ type RuntimeSettingResponse struct {
 	Hidden      bool
 }
 
-// Settings define the runtime settings the component would understand
-type Settings map[string]RuntimeSetting
+// Params that the settings component need
+type Params struct {
+	// Settings define the runtime settings the component would understand
+	Settings map[string]RuntimeSetting
+	// The config object in which the settings would apply operations
+	Config config.Component
```

Comment:
It's often hard to know which config we're working with for runtime settings (core, system-probe, security ...).

What about having a dedicated field for each ?

Reply:
That is fair point. 

As discussed IRL. Today each agent process handles their own configuration, so each setting component needs a single config component to interact. 

In the future when settings configuration is only handled via the core agent we could store each of the config components and change the right config based on the config prefix. Ex: `datadod-agent config set apm_config.log_level` This command would just modify the apm process setting